src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/components/execution-list/execution-card/ExecutionCardBody.tsx

```diff
@@ -0,0 +1,103 @@
+import css from '../ExecutionList.module.scss'
+
+interface ExecutionCardBodyProps {
+  executionHistoryItem: FeaturePipelineExecution
+  flagVariations: Variation[]
+}
+
+const ExecutionCardBody: React.FC<ExecutionCardBodyProps> = ({ flagVariations, executionHistoryItem }) => {
+  const { getVariationColorByName } = useFlagVariation({
+    flagVariations
+  })
+
+  const { getString } = useStrings()
+
+  const percentageRolloutRules =
+    executionHistoryItem.triggerDetails.rules?.filter(rule => rule.serve.distribution) || []
+
+  return (
+    <Container padding="medium" flex={{ justifyContent: 'space-between', alignItems: 'baseline' }} width="100%">
+      <Collapse
+        heading={
+          <Container flex={{ justifyContent: 'space-between' }} width="100%">
+            <Text font={{ variation: FontVariation.BODY2 }} color={Color.GREY_900} margin={{ left: 'small' }}>
+              {getString('cf.featureFlags.flagPipeline.triggerDetails')}
+            </Text>
+          </Container>
+        }
+        collapseClassName={css.collapse}
+      >
+        <div className={css.collapseBody}>
+          {executionHistoryItem.triggerDetails.state && (
+            <>
+              <Text font={{ variation: FontVariation.SMALL_SEMI }} color={Color.GREY_500} inline>
+                {getString('cf.featureFlags.flagPipeline.flagState')}:
+              </Text>
+              <Text
+                font={{ variation: FontVariation.SMALL_SEMI }}
+                color={Color.GREY_900}
+                inline
+                data-testid="flag-state"
+              >
+                {executionHistoryItem.triggerDetails.state}
+              </Text>
+            </>
+          )}
+
+          <Text font={{ variation: FontVariation.SMALL_SEMI }} color={Color.GREY_500} inline>
+            {getString('cf.featureFlags.targeting')}:
+          </Text>
+          {executionHistoryItem.triggerDetails.variationMap?.map(variationMapItem => (
+            <>
+              {variationMapItem.targets && (
+                <TargetItem
+                  targets={variationMapItem.targets}
+                  variation={variationMapItem.variation}
+                  getVariationColorByName={getVariationColorByName}
+                />
+              )}
+              {variationMapItem.targetSegments && (
+                <TargetGroupItem
+                  targetSegments={variationMapItem.targetSegments}
+                  variation={variationMapItem.variation}
+                  getVariationColorByName={getVariationColorByName}
+                />
+              )}
+            </>
+          ))}
+          {percentageRolloutRules.length > 0
+            ? percentageRolloutRules.map(percentageRollout => (
+                <PercentageRolloutItem
+                  key={percentageRollout.ruleId}
+                  percentageRollout={percentageRollout}
+                  flagVariations={flagVariations}
+                />
+              ))
+            : null}
```

Comment:
It should be safe to just do the `map` without the `length` check
```suggestion
          {percentageRolloutRules.map(percentageRollout => (
            <PercentageRolloutItem
              key={percentageRollout.ruleId}
              percentageRollout={percentageRollout}
              flagVariations={flagVariations}
            />
           ))}
```

Reply:
fixed